include/librealsense2/dds/dds-device-server.cpp

```diff
@@ -86,6 +90,171 @@ class dds_device_server::dds_stream_server
     image_header _image_header;
 };
 
+//------------------------------------------------------------------------------------------------------------//
+// DDS notifications server is in charge of publishing notifications and info like streams, profiles, events
+//------------------------------------------------------------------------------------------------------------//
+// We want to know when readers join our topic
+class dds_device_server::dds_notifications_server
+{
+public:
+    // We want to know when readers join our topic
+    class dds_notifications_client_listener : public eprosima::fastdds::dds::DataWriterListener
+    {
+    public:
+        dds_notifications_client_listener( dds_notifications_server * owner )
+            : eprosima::fastdds::dds::DataWriterListener()
+            , _owner( owner )
+        {
+        }
+
+        void on_publication_matched(
+            eprosima::fastdds::dds::DataWriter * writer,
+            const eprosima::fastdds::dds::PublicationMatchedStatus & info ) override
+        {
+            if( info.current_count_change == 1 )
+            {
+                LOG_DEBUG( "DataReader " << writer->guid() << " for topic: " << writer->get_topic()->get_name() << " discovered" );
+                {
+                    {
+                        std::lock_guard< std::mutex > lock( _owner->_notification_send_mutex );
+                        _owner->_new_reader_joined = true;
+                    }
+                    _owner->_send_notification_cv.notify_all();
+                }
+            }
+            else if( info.current_count_change == -1 )
+            {
+                LOG_DEBUG( "DataReader " << writer->guid() << " for topic: " << writer->get_topic()->get_name() <<" disappeared" );
+            }
+            else
+            {
+                LOG_ERROR( std::to_string( info.current_count_change )
+                           << " is not a valid value for on_publication_matched" );
+            }
+        }
+
+    private:
+        dds_notifications_server * _owner;
+    };
+
+    dds_notifications_server( eprosima::fastdds::dds::DomainParticipant * participant,
+                              eprosima::fastdds::dds::Publisher * publisher,
+                              const std::string & topic_root )
+        : _participant( participant )
+        , _publisher( publisher )
+        , _topic( nullptr )
+        , _data_writer( nullptr )
+        , _clients_listener( this )
+        , _notifications_loop( [this]( dispatcher::cancellable_timer timer ) {
+            if( _active )
+            {
+                // We can send 2 types of notifications:
```

Comment:
I think I finally understand how you structured this:
"latched" are initialization broadcasts we're assuming do not change over time, so are generated once and stored for resend every time a reader joins?
The other messages are one-off messages that are likely (going to be) responses to control commands?

Reply:
Too bad I didn't read this comment before I answered above :)
You understood correctly.